### PR TITLE
Add account uuid extraction middleware

### PR DESCRIPTION
**Pull request by kulmann**
Adds a middleware for extracting the account uuid from the x-access-token header and providing it in the context. This can be used in any ocis service that needs the account uuid. It is minted into the x-access-token by the proxy.

There are version incompatibilities that need to be solved... thus creating this as a draft.

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- account/option.go  1
- middleware/account.go  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-pkg/pullRequest?prid=5586556&bid=18307825)

File: middleware/account.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+				return
+			}
+
+			user, err := tokenManager.DismantleToken(r.Context(), token)
+			if err != nil {
+				opt.Logger.Error().Err(err)
+				return
+			}
+
+			ctx := context.WithValue(r.Context(), UUIDKey, user.Id.OpaqueId)
```
Comment (refs) [inline middleware/account.go:49]: The only piece of feedback I have is that we're tying our data model to Reva here for all consumers of this context. Should we instead deal with the UUID, and not missleadingly store an opaque id under a `UUIDKey`?

**Review by refs — APPROVED** [review]

File: middleware/account.go (hunk 1 above)
Comment (kulmann) [reply middleware/account.go:49]: The opaque id is what we previously set in the proxy and is in fact our UUID. Maybe a good idea to have the proxy middleware about the account uuid here in ocis-pkg as well, to make it more transparent what we're doing.

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- account/option.go  1
- middleware/account.go  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-pkg/pullRequest?prid=5586556&bid=18307825)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- account/option.go  1
- middleware/account.go  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-pkg/pullRequest?prid=5586556&bid=18307825)

**Pull request merged**